go/storage/api/context.go

```diff
@@ -45,3 +52,47 @@ func NodePriorityHintFromContext(ctx context.Context) []signature.PublicKey {
 	nodes, _ := ctx.Value(contextKeyNodePriorityHint).([]signature.PublicKey)
 	return nodes
 }
+
+// WithNodeBlacklist sets a storage blacklist key for any storage requests using this context.
+// The blacklist is initially empty (i.e. all nodes are acceptable).
+func WithNodeBlacklist(ctx context.Context) context.Context {
+	return context.WithValue(ctx, contextKeyNodeBlacklist, map[signature.PublicKey]struct{}{})
+}
+
+// BlacklistAddNode adds a node to the blacklist associated with the context. If there's no
+// associated blacklist, the function does nothing.
+func BlacklistAddNode(ctx context.Context, node *node.Node) {
```

Comment:
Instead of doing it like this, can we make a blacklist interface that has a method for checking whether something is blacklisted? And then you can configure an instance using `WithNodeBlacklist`?